pydotthz/pydotthz.py

```diff
@@ -85,305 +84,331 @@ def add_field(self, key, value):
 
-       This class is designed to store `DotthzMeasurement` objects and ensure that
-       whenever a new measurement is added or modified, the corresponding measurement
-       is written to the file using the `write_measurement()` method.
+    This class provides persistent storage of THz time-domain spectroscopy data via HDF5.
 
-       Inherits from `dict` to retain normal dictionary behavior (e.g., iteration,
-       item retrieval), while adding the functionality of writing measurements to
-       an HDF5 file automatically.
+    Supports context manager (`with` statement) for automatic file handling.
+    """
 
-       Attributes
-       ----------
-       _file : DotthzFile
-           A reference to the `DotthzFile` instance to which measurements will be written.
+    def __init__(self, name: Union[str, Path], mode="r", driver=None, libver=None,
+                 userblock_size=None, swmr=False, rdcc_nslots=None,
+                 rdcc_nbytes=None, rdcc_w0=None, track_order=None,
+                 fs_strategy=None, fs_persist=False, fs_threshold=1,
+                 fs_page_size=None, page_buf_size=None, min_meta_keep=0,
+                 min_raw_keep=0, locking=None, alignment_threshold=1,
+                 alignment_interval=1, meta_block_size=None, **kwds):
+        self.file = h5py.File(name, mode, driver=driver, libver=libver,
+                              userblock_size=userblock_size, swmr=swmr,
+                              rdcc_nslots=rdcc_nslots, rdcc_nbytes=rdcc_nbytes,
+                              rdcc_w0=rdcc_w0, track_order=track_order,
+                              fs_strategy=fs_strategy, fs_persist=fs_persist,
+                              fs_threshold=fs_threshold,
+                              fs_page_size=fs_page_size,
+                              page_buf_size=page_buf_size,
+                              min_meta_keep=min_meta_keep,
+                              min_raw_keep=min_raw_keep, locking=locking,
+                              alignment_threshold=alignment_threshold,
+                              alignment_interval=alignment_interval,
+                              meta_block_size=meta_block_size, **kwds)
 
-       Methods
-       -------
-       __setitem__(self, key, value):
-           Adds a new measurement to the dictionary and automatically writes it to the file.
+    def __enter__(self):
+        # Enable the use of the `with` statement.
+        return self
 
-       __init__(self, base_file, *args, **kwargs):
-           Initializes the `MeasurementDict` and stores a reference to the parent `DotthzFile`.
-       """
+    def __exit__(self, exc_type, exc_value, traceback):
+        # Close any resources if applicable.
+        if self.file is not None:
+            self.file.close()
+            self.file = None
 
-    def __init__(self, base_file, *args, **kwargs):
-        """
-        Initializes the `MeasurementDict` object.
+    def __getitem__(self, key):
```

Comment:
Since we're moving to a Dict-like structure I suggest we support creating measurements at assignment. This can be done easily by modifying the __getitem__ method.

```suggestion
    def __getitem__(self, key):
        # Create a group if it doesn't already exist.
        if key not in self.keys():
            self.create_group(key)

        # Return a DotthzMeasurementWrapper for the group
        return DotthzMeasurementWrapper(self.file[key])
```